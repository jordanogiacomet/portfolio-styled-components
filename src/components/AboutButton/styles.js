import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
