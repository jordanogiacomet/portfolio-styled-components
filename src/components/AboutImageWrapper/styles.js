import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
