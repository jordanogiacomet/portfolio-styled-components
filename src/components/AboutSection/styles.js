import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
