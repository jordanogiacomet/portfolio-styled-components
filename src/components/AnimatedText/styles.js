import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;