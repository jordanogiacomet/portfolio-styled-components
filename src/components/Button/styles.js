import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 