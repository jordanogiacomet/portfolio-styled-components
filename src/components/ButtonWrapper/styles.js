import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;