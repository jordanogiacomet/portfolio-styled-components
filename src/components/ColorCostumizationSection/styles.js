// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
