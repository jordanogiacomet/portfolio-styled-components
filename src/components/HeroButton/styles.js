import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px;
    transition: scale ease-in-out 0.2s;
    border: transparent;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };
    @media (${theme.media.lteMedium}) {
      width: 100%;
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
  ${({ from, via, to}) => css`
    background: linear-gradient(to bottom right, ${from}, ${via}, ${to});
  `}
`; 
