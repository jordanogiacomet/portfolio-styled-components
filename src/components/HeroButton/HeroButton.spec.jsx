import { renderTheme } from '../../styles/render-theme';
import { HeroButton } from '.';
import { screen } from '@testing-library/react';
import { theme } from '../../styles/theme';

describe('<HeroButton />', () => {
  it('should render button with children', () => {
    renderTheme(<HeroButton>Click</HeroButton>);
    const button = screen.getByText('Click');
    expect(button).toBeInTheDocument();
  });

  it('should apply default button styles', () => {
    const { container } = renderTheme(<HeroButton>Click</HeroButton>);
    expect(container.firstChild).toHaveStyleRule('padding', theme.button.default.padding);
    expect(container.firstChild).toHaveStyleRule('border-radius', theme.button.default.borderRadius);
    expect(container.firstChild).toHaveStyleRule('color', theme.button.default.textColor);
    expect(container.firstChild).toHaveStyleRule('display', 'inline-flex');
    expect(container.firstChild).toHaveStyleRule('align-items', 'center');
    expect(container.firstChild).toHaveStyleRule('justify-content', 'center');
    expect(container.firstChild).toHaveStyleRule('min-width', '150px');
  });

  it('should apply hireMe button styles', () => {
    const { container } = renderTheme(<HeroButton type="hireMe">Hire me</HeroButton>);
    expect(container.firstChild).toHaveStyleRule('padding', theme.button.hireMe.padding);
    expect(container.firstChild).toHaveStyleRule('border-radius', theme.button.hireMe.borderRadius);
    expect(container.firstChild).toHaveStyleRule('color', theme.button.hireMe.textColor);
  });

  it('should be accessible with an aria-label', () => {
    renderTheme(<HeroButton aria-label="accessible button">Accessible</HeroButton>);
    expect(screen.getByLabelText('accessible button')).toBeInTheDocument();
  });
});
