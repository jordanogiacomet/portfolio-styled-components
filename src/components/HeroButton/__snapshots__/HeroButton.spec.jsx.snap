// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroButton /> should match snapshot 1`] = `
.c0 {
  padding: 1rem 1.5rem;
  border-radius: 9999px;
  color: #FFFFFF;
  font-weight: bold;
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  min-width: 150px;
  transition: scale ease-in-out 0.2s;
  border: transparent;
  background: linear-gradient(to bottom right, #1E90FF, #4169E1, #00008B);
}

.c0:hover {
  transition: scale ease-in-out 0.2s;
  scale: calc(1.03);
}

@media ((max-width: 768px)) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0 dark"
    from="#1E90FF"
    to="#00008B"
    type="default"
    via="#4169E1"
  >
    Me Contrate
  </button>
</div>
`;
